### PR TITLE
Eliminate handle_zero_dim when broadcasting is applied earlier.

### PR DESCRIPTION
**Pull request by gchanan**
This ends up not actually doing anything unless all the broadcasted tensors are scalars,
which ends up with inconsistent behavior in that case only, because the type promotion rules are different.

This is better solved with real type promotion logic.

### CODE REVIEW

**Comment by gchanan** [comment]
@apaszke

File: aten/src/ATen/function_wrapper.py [hunk 1]
```diff
@@ -1079,15 +1079,19 @@ def is_actual_return_long(ret):
...
+        zero_dim_dispatch = option.get('zero_dim_dispatch_when_scalar', '')
         if not zero_dim_dispatch:
             return []
+        broadcasts_arg = zero_dim_dispatch in option.get('broadcast_actuals', '')
+        zero_dim_only = option.get('zero_dim_tensor_only', False)
+        # this combination doesn't seem to make sense
+        assert not (broadcasts_arg and zero_dim_only)
+        # if the argument broadcasts, then this would only affect cases where all broadcasted
+        # tensors were zero-dim, which is inconsistent with the scalar handling.
+        if broadcasts_arg:
```
Comment (apaszke) [inline aten/src/ATen/function_wrapper.py]: Is this ever false?

**Review by apaszke — APPROVED** [review]
LGTM but IIRC there are no cases where we would actually generate this code right now. It might be better to remove this code altogether

File: aten/src/ATen/function_wrapper.py (hunk 1 above)
Comment (gchanan) [reply aten/src/ATen/function_wrapper.py]: It is used for `scatter` which allows you to pass zero-dim tensors and for them to behave like scalars.

**Comment by gchanan** [comment]
This change seems to break the jit tests, although I guess the jit only worked previously on those specific conditions (lhs and rhs are both scaalrs).

**Comment by apaszke** [comment]
Just change the input tensors in that test from float to long. I wonder why only a single CI configuration failed, it should be deterministic...

**Review by apaszke — APPROVED** [review]

**Pull request merged**